en/for-ent.md

```diff
@@ -0,0 +1,190 @@
+
+### Prerequisites
+Make sure you have [Go](https://golang.org/doc/install) installed (1.17 or above).  
+Create a folder for this demo, and initialize using `go mod`:
+
+{{< copyable "" >}}
+
+```bash
+mkdir todo
+cd todo
+go mod init todo
+```
+### Initialize project
+First, we need to install Ent and initialize the project structure. run:
+
+{{< copyable "" >}}
+
+```bash
+go run -mod=mod entgo.io/ent/cmd/ent init Todo
+```
+After running the above, your project directory should look like this:
+```bash
+.
+├── ent
+│   ├── generate.go
+│   └── schema
+│       └── todo.go
+├── go.mod
+└── go.sum
+```
+Open `schema/todo.go` and add some fields to the `Todo` entity:
+
+{{< copyable "" >}}
+
+```go
+func (Todo) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("title"),
+		field.String("content"),
+		field.Bool("done").
+			Default(false),
+		field.Time("created_at"),
+	}
+}
+```
+Finally, run:
+
+{{< copyable "" >}}
+
+```bash
+go generate ./ent
+```
+
+### Run the example application
+Create a file named `main.go` in your project's root folder and copy to it the the following code:
+
+{{< copyable "" >}}
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+	"todo/ent"
+	"todo/ent/todo"
+
+	"entgo.io/ent/dialect/sql/schema"
+	_ "github.com/go-sql-driver/mysql"
+)
+
+func main() {
+	// Connect to TiDB.
+	client, err := ent.Open("mysql", "root@tcp(127.0.0.1:4000)/entgo?parseTime=true")
+	if err != nil {
+		log.Fatal("error opening ent client", err)
+	}
+	defer client.Close()
+	// Migrate the local schema with the DB.
+	if err := client.Schema.Create(
+		context.Background(),
+		// TiDB support is possible with Atlas engine only.
+		schema.WithAtlas(true),
+	); err != nil {
+		log.Fatal("error migrating DB", err)
+	}
+
+	// Create some todos
+	client.Todo.Create().
+		SetTitle("buy groceries").
+		SetContent("tomato, lettuce and cucumber").
+		SetCreatedAt(time.Now()).
+		SaveX(context.Background())
+	client.Todo.Create().
+		SetTitle("See my doctor").
+		SetContent("periodic blood test").
+		SetCreatedAt(time.Now()).
+		SaveX(context.Background())
+
+	// query todo with 'where' filter
+	todos := client.Todo.Query().
+		Where(todo.ContentContains("tomato")).
+		AllX(context.Background())
+	fmt.Printf("The following todos contain 'tomato' in their content: %v\n", todos)
+
+	// after buying the groceries, set the todo as done.
+	client.Todo.UpdateOne(todos[0]).SetDone(true).ExecX(context.Background())
+
+	// delete todos that are done.
+	deletedCount := client.Todo.Delete().Where(todo.Done(true)).ExecX(context.Background())
+	fmt.Printf("deleted %d todos\n", deletedCount)
+}
+```
+### Run the code
+
+Run the `main.go` code:
+
+{{< copyable "" >}}
+
+```bash
+go run main.go
+```
+
+The expected output is as follows:
+```
+The following todos contain 'tomato' in their content:
+[Todo(id=1, title=buy groceries, content=tomato, lettuce and cucumber, done=false, created_at=Mon May  2 14:32:20 2022)]
+deleted 1 todos
+```
```

Comment:
I think we have more information to tell the users. For exmaple, Tidb deosn;t support the function. we can add some link to those documents.

Reply:
@hawkingrei I think that TiDB features support is out of the scope of this example and there's no use for savepoints here.  
> more in-depth experience of using ent  

What do you mean by that? can you please share an example from a different doc?

Reply:
You can add some links which look like it.

<img width="468" alt="image" src="https://user-images.githubusercontent.com/3427324/169990733-ae21f474-77b4-4ede-9ac5-479f792bedfd.png">

for example, you can add a link like [this](https://entgo.io/docs/migrate).

BTW, I find the other document about orm quick start having this party. I will improve it as soon as possible.

Reply:
Thanks for the example. I added some links about more advanced topics in Ent.  
Please feel free to add links to relevant TiDB topics.